extensions/wikia/PhalanxII/classes/Phalanx.class.php

```diff
@@ -117,48 +130,128 @@ public function load() {
 				'reason'    => $row->p_reason,
 				'comment'   => $row->p_comment,
 				'ip_hex'    => $row->p_ip_hex
-			);
+			];
 		}
 
 		wfProfileOut( __METHOD__ );
 	}
 
+	public function isTypeContent() {
+		return $this->data['type'] & self::TYPE_CONTENT;
+	}
+
+	public function isTypeSummary() {
+		return $this->data['type'] & self::TYPE_SUMMARY;
+	}
+
+	public function isTypeTitle() {
+		return $this->data['type'] & self::TYPE_TITLE;
+	}
+
+	public function isTypeUser() {
+		return $this->data['type'] & self::TYPE_USER;
+	}
+
+	public function isTypeQuestionTitle() {
+		return $this->data['type'] & self::TYPE_ANSWERS_QUESTION_TITLE;
+	}
+
+	public function isTypeRecentQuestions() {
+		return $this->data['type'] & self::TYPE_ANSWERS_RECENT_QUESTIONS;
+	}
+
+	public function isTypeWikiCreation() {
+		return $this->data['type'] & self::TYPE_WIKI_CREATION;
+	}
+
+	public function isTypeEmail() {
+		return $this->data['type'] & self::TYPE_EMAIL;
+	}
+
 	public function save() {
 		wfProfileIn( __METHOD__ );
 
-		if ( ( $this->data['type'] & self::TYPE_USER ) && User::isIP( $this->data['text'] ) ) {
-			if ( wfIsTrustedProxy( $this->data['text'] ) ) {
-				// Don't allow to set blocks for trusted proxies or Wikia network hosts
-				return false;
-			}
-
-			$this->data['ip_hex'] = IP::toHex( $this->data['text'] );
+		if ( !$this->validateUserFilter() ) {
+			return false;
 		}
 
-		$dbw = $this->getSharedDB( DB_MASTER );
+		$dbw = $this->getDatabase( DB_MASTER );
 		if ( empty( $this->data['id'] ) ) {
 			/* add block */
 			$dbw->insert( $this->db_table, $this->mapToDB(), __METHOD__ );
-			$action = 'add';
+			$action = static::ACTION_ADD_FILTER;
 		} else {
 			$dbw->update( $this->db_table, $this->mapToDB(), array( 'p_id' => $this->data['id'] ), __METHOD__ );
-			$action = 'edit';
+			$action = static::ACTION_MODIFY_FILTER;
 		};
 
 		if ( $dbw->affectedRows() ) {
-			if ( $action == 'add' ) {
+			if ( $action === static::ACTION_ADD_FILTER ) {
 				$this->data['id'] = $dbw->insertId();
 			}
 			$this->log( $action );
 		} else {
 			$action = '';
 		}
+		// Commit the transaction so that Phalanx service will see the update when it is notified
 		$dbw->commit();
 
 		wfProfileOut( __METHOD__ );
 		return ( $action ) ? $this->data['id'] : false;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)